src/devices/wec2103.c

```diff
@@ -0,0 +1,95 @@
+/** @file
+    WEC-2103 temperature/humidity sensor
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/*
+-X 'n=name,m=OOK_PPM,s=2000,l=4000,g=4400,r=9400'
+
+Similar to prologue, kedsum, esperanza_ews, s3318p
+Only available information for this device: https://fcc.report/FCC-ID/WEC-2103
+
+Example datagram:
+ f2 90          6b5        96       1     8
+|ID|Checksum?|Temperature|Humidity|Channel|
+
+Temperature in Fahrenheit*100+900->hex
+Example: 82.4F->824->1724->0x6bc
+
+
+Byte:      0        1        2        3        4        5
+Nibble:    1   2    3   4    5   6    7   8    9   10   11
+Type:      IIIIIIII XXXXFFFF TTTTTTTT TTTTHHHH HHHHCCCC ????
+
+I: random device ID, changes on powercycle
+X: Checksum?
+F: Flags
+T: Temperature
+H: Humidity
+
+Flags: tx-button pressed|?|?|?
+
+*/
+
+#include "decoder.h"
+
+static int wec2103_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t b[6];
+    data_t *data;
+
+    if (bitbuffer->num_rows != 6  || bitbuffer->bits_per_row[2] != 42)
+        return DECODE_ABORT_LENGTH;
+
+    bitbuffer_extract_bytes(bitbuffer, 3, 0, b, 42);
+
+    int temp_raw  = (b[2] << 4) | ((b[3] & 0xf0) >> 4);
+
+    int device_id = b[0];
+    int channel   = b[4] & 0x0f;
+    int flags = b[1] & 0xf;
+    float temp_f  = (temp_raw - 900) * 0.1f;
+    int humidity  = ((b[3] & 0x0f) * 10) + ((b[4] & 0xf0) >> 4);
+    int button   = (b[1] & 0x08) >> 3;
+
+    data = data_make(
```

Comment:
Surround with `/* clang-format off */` `/* clang-format on */` as seen in other decoders.